thunder/core/utils.py

```diff
@@ -1123,3 +1124,13 @@ def partition(pred, iterable):
     # partition(is_odd, range(10)) --> 0 2 4 6 8   and  1 3 5 7 9
     t1, t2 = itertools.tee(iterable)
     return itertools.filterfalse(pred, t1), filter(pred, t2)
+
+
+def make_hashable(item: Any, /) -> tuple:
+    if isinstance(item, (list, tuple)):
+        return tuple(map(make_hashable, item))
+    if isinstance(item, dict):
+        return FrozenDict(item)
+    if isinstance(item, slice):
+        return id(item)
```

Comment:
What function uses slice objects as inputs? What do you think about checking if the object is hashable or not and taking its id?

Reply:
Yeah I was thinking the same, as of now also some other libraries check for specific types before deciding is things are hashable or not.  However, I am looking out for types that can help with that if there are any and I think that would be the cleaner option